src/main/java/io/supertokens/inmemorydb/queries/JWTSigningQueries.java

```diff
@@ -16,20 +16,93 @@
+import io.supertokens.pluginInterface.jwt.JWTSymmetricSigningKeyInfo;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
 
 public class JWTSigningQueries {
+    private final static String lockString = "jwt_signing_keys_table_lock";
+
     static String getQueryToCreateJWTSigningTable(Start start) {
         return "CREATE TABLE IF NOT EXISTS " + Config.getConfig(start).getJWTSigningKeysTable() + " ("
-                + "key_id VARCHAR(255) NOT NULL," + "public_key TEXT NOT NULL,"
-                + "private_key TEXT NOT NULL," + "algorithm VARCHAR(10) NOT NULL,"
-                + "algorithm_type VARCHAR(10) NOT NULL," + "created_at BIGINT UNSIGNED,"
+                + "key_id VARCHAR(255) NOT NULL," + "key_string TEXT NOT NULL,"
+                + "algorithm VARCHAR(10) NOT NULL," + "created_at BIGINT UNSIGNED,"
                 + "PRIMARY KEY(key_id));";
     }
 
-    static String getQueryToFetchLatestKey(Start start) {
-        return "SELECT * FROM " + Config.getConfig(start).getJWTSigningKeysTable()
-                + "ORDER BY created_at DESC";
+    public static List<JWTSigningKeyInfo> getJWTSigningKeys_Transaction(Start start, Connection con)
+            throws SQLException, StorageQueryException {
+
+        ((ConnectionWithLocks) con).lock(lockString);
+
+        String QUERY = "SELECT * FROM "
+                + Config.getConfig(start).getJWTSigningKeysTable()
+                + " ORDER BY created_at DESC;";
+
+        try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+            ResultSet result = pst.executeQuery();
+            List<JWTSigningKeyInfo> keys = new ArrayList<>();
+
+            while(result.next()) {
+                keys.add(JWTSigningKeyInfoRowMapper.getInstance().mapOrThrow(result));
+            }
+
+            return keys;
+        }
+    }
+
+    private static class JWTSigningKeyInfoRowMapper implements RowMapper<JWTSigningKeyInfo, ResultSet> {
+        private static final JWTSigningKeyInfoRowMapper INSTANCE = new JWTSigningKeyInfoRowMapper();
+
+        private JWTSigningKeyInfoRowMapper() {
+        }
+
+        private static JWTSigningKeyInfoRowMapper getInstance() {
+            return INSTANCE;
+        }
+
+        @Override
+        public JWTSigningKeyInfo map(ResultSet result) throws Exception {
+            String keyId = result.getString("key_id");
+            String keyString = result.getString("key_string");
+            long createdAt = result.getLong("created_at");
+            String algorithm = result.getString("algorithm");
+
+            if (keyString.contains("|")) {
+                return JWTAsymmetricSigningKeyInfo.withKeyString(keyId, createdAt, algorithm, keyString);
+            } else {
+                return new JWTSymmetricSigningKeyInfo(keyId, createdAt, algorithm, keyString);
```

Comment:
Why is one using `withKeyString` and the other just calling the constructor?

Reply:
For Symmetric keys the key string is the key and does not need any extra computation, for Asymmetric keys its a helper function that splits the keystring with the "|" delimiter and then instantiates using the class constructor

Reply:
But can't the constructor of the `JWTAsymmetricSigningKeyInfo` always just do that? You can have two constructors in that class - one that accepts just a string, and one that accepts the public and private key (same as the existing constructor)

Reply:
I can do that as well, I just prefer having static methods. But for the sake of consistency probably a better idea to go with multiple constructors, ill make the change

Reply:
Done (Updated plugin interface PR https://github.com/supertokens/supertokens-plugin-interface/pull/22)